internal/component/prometheus/scrape/scrape.go

```diff
@@ -252,12 +260,57 @@ func New(o component.Options, args Arguments) (*Component, error) {
 		opts:                o,
 		cluster:             clusterData,
 		reloadTargets:       make(chan struct{}, 1),
-		scraper:             scraper,
+		debugDataPublisher:  debugDataPublisher.(livedebugging.DebugDataPublisher),
 		appendable:          alloyAppendable,
 		targetsGauge:        targetsGauge,
 		movedTargetsCounter: movedTargetsCounter,
 		unregisterer:        unregisterer,
 	}
+	componentID := livedebugging.ComponentID(c.opts.ID)
+	interceptor := prometheus.NewInterceptor(alloyAppendable, ls,
+		prometheus.WithAppendHook(func(globalRef storage.SeriesRef, l labels.Labels, t int64, v float64, next storage.Appender) (storage.SeriesRef, error) {
+			_, nextErr := next.Append(globalRef, l, t, v)
+			if c.debugDataPublisher.IsActive(componentID) {
+				c.debugDataPublisher.Publish(componentID, fmt.Sprintf("ts=%d, labels=%s, value=%f", t, l, v))
```

Comment:
```suggestion
				c.debugDataPublisher.Publish(componentID, fmt.Sprintf("ts=%d, type=histogram, labels=%s, value=%f", t, l, v))
```
I think each hook should print its type so it's easier to read this